python/pylibcugraph/pylibcugraph/replication.pyx

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+
+from pylibcugraph._cugraph_c.resource_handle cimport (
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_free,
+)
+from pylibcugraph._cugraph_c.graph_functions cimport (
+    cugraph_allgather,
+    cugraph_induced_subgraph_result_t,
+    cugraph_induced_subgraph_get_sources,
+    cugraph_induced_subgraph_get_destinations,
+    cugraph_induced_subgraph_get_edge_weights,
+    cugraph_induced_subgraph_get_edge_ids,
+    cugraph_induced_subgraph_get_edge_type_ids,
+    cugraph_induced_subgraph_get_subgraph_offsets,
+    cugraph_induced_subgraph_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    assert_CAI_type,
+    copy_to_cupy_array,
+    create_cugraph_type_erased_device_array_view_from_py_obj
+)
+
+
+def replication(ResourceHandle resource_handle,
```

Comment:
I think a better name would be `replicate_edgelist`, same for the file itself.

Reply:
Done